src/psyclone/gocean1p0.py

```diff
@@ -1553,17 +1558,21 @@ def __init__(self, arg):
 
     @property
     def name(self):
-        ''' Returns the Fortran name of the grid property. This name is
-            used in the generated code like so: <fld>%grid%name '''
+        ''' Returns the Fortran name of the grid property, which is used
+        in error messages etc.'''
         return self._name
 
-    @property
-    def dereference_name(self):
-        ''':returns: the dereference string required to access a property in\
-        a dl_esm field (e.g. "subdomain%internal%xstart"). This name is\
-        used in the generated code like so: <fld>%grid%dereference_name
+    def dereference(self, fld_name):
+        '''Returns a Fortran string to dereference a property of the
+        specified field. E.g."name%grid%dx". The stored value of
+        self._dereference_name is a format string, where {0} represents
+        the field name.
+
+        :returns: the dereference string required to access a property in \
```

Comment:
This is inside a grid-argument class so this should say "access a grid property"?

Reply:
Done.